worker/operatorhttpserver/server.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package operatorhttpserver
+
+import (
+	"crypto/tls"
+	"net/http"
+
+	"github.com/juju/errors"
+	"gopkg.in/juju/worker.v1/catacomb"
+
+	"github.com/juju/juju/apiserver/apiserverhttp"
+	"github.com/juju/juju/pki"
+)
+
+type Logger interface {
+	Errorf(string, ...interface{})
+	Infof(string, ...interface{})
+}
+
+type Server struct {
+	catacomb catacomb.Catacomb
+	logger   Logger
+	Mux      *apiserverhttp.Mux
+	server   *http.Server
+}
+
+func (s *Server) Kill() {
+	s.catacomb.Kill(nil)
+}
+
+func (s *Server) loop() error {
+	httpCh := make(chan error)
+
+	go func() {
+		s.logger.Infof("starting operator http server on %s", s.server.Addr)
+		httpCh <- s.server.ListenAndServeTLS("", "")
+		close(httpCh)
+	}()
+
+	select {
```

Comment:
We might be missing some shutdown logic - see httpserver worker

Reply:
Try that